### PR TITLE
bench: fix running the crypto bench + add back sled to compare sled with sqlite

### PR DESCRIPTION
**Pull request by bnjbvr**
- This fixes the crypto bench by ensuring a tokio runtime is active whenever a `OlmMachine` built with a `SqliteCryptoStore` is dropped. That's necessary because `deadpool` makes use of `block_on` in the `Drop` implementation of one of its managers.
- This also adds back sled to the crypto benchmarks, so we can compare the performance of the two backends.

Turns out that sled was faster than sqlite: full report temporarily hosted on https://benj.me/pub/criterion/report/

# TL;DR: median run times

| Benchmark | Mem | Sled | Sqlite |
| -- | -- | -- | -- |
| devices missing session collecting | 8.59 ms | 16.23 ms | 133.36 ms |
| keys quering |3.5 ms | 6.69 ms | 8.01 ms |
| olm session creation | 35.24 ms | 38.07 ms | 46.74 ms |
| room key sharing | 2 ms | 5.18 ms | 9.27 ms |

### CODE REVIEW

**Comment by poljar** [comment]
Nice, this gives us a clue that `devices missing session collecting` on `SQLite` might be not optimized enough. The rest seems fine.

Good thing we double checked, and thanks for hosting the full report.

**Review by Hywan — APPROVED** [review]
It looks good to me, thanks!

Next step is to fix the SQLite store, as a separate PR.
